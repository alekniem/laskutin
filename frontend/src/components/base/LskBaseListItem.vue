<template>
    <v-list-item :id="listItemId">
        <v-list-item-content
            class="text-caption blue--text text--darken-2 font-weight-bold"
        >
            {{ text | trans }}
        </v-list-item-content>
        <v-list-item-content>
            {{ value }}
        </v-list-item-content>
    </v-list-item>
</template>

<script>
import camelCase from "lodash/camelCase";
import upperFirst from "lodash/upperFirst";

export default {
    name: "LskBaseListItem",

    props: {
        itemId: { type: String, required: true },
        parentId: { type: String, required: true },
        text: { type: String, required: true },
        value: { type: [String, Number], required: true },
    },

    computed: {
        listItemId() {
            return (
                this.parentId + upperFirst(camelCase(this.itemId)) + "ListItem"
            );
        },
    },
};
</script>
