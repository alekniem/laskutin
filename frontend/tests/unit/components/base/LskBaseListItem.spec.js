import LskBaseListItem from "@/components/base/LskBaseListItem";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskBaseListItem", () => {
    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseListItem, {
            propsData: {
                itemId: "TEST_ITEM_ID",
                parentId: "TEST_PARENT_ID",
                text: "TEST TEXT",
                value: "TEST VALUE",
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
