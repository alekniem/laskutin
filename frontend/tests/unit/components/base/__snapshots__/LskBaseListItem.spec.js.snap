// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskBaseListItem should match snapshot 1`] = `
<div tabindex="-1"
     id="TEST_PARENT_IDTestItemIdListItem"
     class="v-list-item theme--light"
>
  <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
    TEST TEXT
  </div>
  <div class="v-list-item__content">
    TEST VALUE
  </div>
</div>
`;
